### DISCUSSION TITLE
Modifying prompt on af-magic theme

### DISCUSSION THREAD

**Discussion by marktsavage** [ohmyzsh/ohmyzsh#10577] [Help] [answered]
Hello there,

I'm totally new to oh-my-zsh (and zsh shell) and loving it so far. I'm currently using the af-magic theme.

I've only basic shell scripting experience, so I'm wondering how I can customise / modify the prompt?

Currently it is displaying the full directory path, which can get very long. I'd like to only show the current directory rather than full path.
I've gone through the [prompt expansion](https://zsh.sourceforge.io/Doc/Release/Prompt-Expansion.html) docs, but any pointers or help much appreciated.

Mark

**Comment by mcornella** [comment, marked as answer]
The `af-magic` theme uses `%~` to show the current directory while substituting your home directory with `~`. You can tell zsh how many path segments to show by putting a number between `%` and `~`: `%1~` will only show 1 path segment (the current directory name).

If that's what you want, you can modify this line https://github.com/ohmyzsh/ohmyzsh/blob/c172624398918577bcb212d07b0036f4058cb5c6/themes/af-magic.zsh-theme#L22
and replace `%~` for `%1~`.

Note that first you have to set it up as a custom theme. See [the guide here](https://github.com/ohmyzsh/ohmyzsh/wiki/Customization#overriding-and-adding-themes).

----
If you instead want to shrink the previous path segments you could take a look at the [`shrink-path` plugin](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/shrink-path).

**Reply by marktsavage** [reply]
Many thanks, that's perfect. I misread the docs, was looking for a different combination of characters!
